erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1591,3 +1591,45 @@ def create_invoice_discounting(source_name, target_doc=None):
 def get_item_details(item_code):
 	item = frappe.get_doc("Item", item_code).as_dict()
 	return item
+
+@frappe.whitelist()
+def email_coa(docname):
+	"""Email COA's from sales invoice
+	Args:
+		docname (str): Sales Invoice Document name
+	Returns:
+		success : if Certificates of Analysis are mailed successfully
+	"""
+	sales_invoice = frappe.get_doc("Sales Invoice", docname)
+	attachments = []
+
+ 	# get customer emails to send emails 
+	contact_person_email = frappe.db.get_value("Customer", sales_invoice.customer, "email_id")
+
+	if not contact_person_email:
+		frappe.throw(_("No contact email found in customer {0} .").format(sales_invoice.customer))
+	
+	for item in sales_invoice.items:
+		# checks certificate of analysis for each item in Sales Invoice Item List
+		coas = frappe.db.get_value("Batch", item.batch_no, 'certificate_of_analysis')
+
+		if not coas:
+			frappe.throw(_("No Certificates of Analysis attached for Item {0} .").format(item.item_code))
```

Comment:
```suggestion
			frappe.throw(_("No Certificates of Analysis attached for Item {0} .").format(item.item_code))
```
- `frappe.throw` will stop the code execution. `continue` after that will have no effect
- Change it to `frappe.msgprint`.

Reply:
Changes done